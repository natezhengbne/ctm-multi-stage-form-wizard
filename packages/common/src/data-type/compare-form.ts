export type CompareForm = {
  personal?: Personal;
  address?: Address;
};

export type Personal = {
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
};

export type Address = {
  streetNumber: string;
  streetName: string;
  streetType: typeof streetTypes[number];
  suburb: string;
  postcode: string;
};

export const streetTypes = [
  "CI", "Ct", "St", "PI", "Ave",
]
