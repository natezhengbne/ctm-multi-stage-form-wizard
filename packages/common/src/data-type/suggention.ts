export type SuburbSuggestions = {
  _embedded: {
    suggestions: Suburb[];
  };
};

export type Suburb = {
  display: {
    text: string;
  };
  source: {
    postcode?: string;
    name?: string;
    state?: string;
  };
};
