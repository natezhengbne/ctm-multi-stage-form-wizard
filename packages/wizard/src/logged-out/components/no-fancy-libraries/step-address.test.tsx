import { CompareForm } from "@ctm-multi-stage-wizard/common/data-type/compare-form";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import StepAddress from "./step-address";

describe("The page of step-address renders", () => {
  it("renders correctly", () => {
    let wrapper = shallow(
      <StepAddress step={1} activeStep={1} onNext={jest.fn()} onPrev={jest.fn()} value={undefined} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("page not active", () => {
    let wrapper = shallow(
      <StepAddress step={1} activeStep={3} onNext={jest.fn()} onPrev={jest.fn()} value={undefined} />
    );
    expect(wrapper.html()).toEqual(null);
  });

  test("all form fields", () => {
    let wrapper = shallow(
      <StepAddress step={1} activeStep={1} onNext={jest.fn()} onPrev={jest.fn()} value={undefined} />
    );

    expect(wrapper.find('input[id="streetNumber"]').props()).toEqual(
      expect.objectContaining({
        required: true,
        name: "streetNumber",
        type: "number",
      })
    );
    expect(wrapper.find('input[id="streetName"]').props()).toEqual(
      expect.objectContaining({
        required: true,
        name: "streetName",
      })
    );
    expect(wrapper.find('select[id="streetType"]').props()).toEqual(
      expect.objectContaining({
        required: true,
        name: "streetType",
      })
    );
    expect(wrapper.find('input[id="suburb"]').props()).toEqual(
      expect.objectContaining({
        required: true,
        name: "suburb",
      })
    );
    expect(wrapper.find('input[id="postcode"]').props()).toEqual(
      expect.objectContaining({
        required: true,
        name: "postcode",
      })
    );
    expect(wrapper.find('button[type="submit"]').length).toEqual(1);
  });
});

describe("The page of step-personal submit form", () => {
  let wrapper: ShallowWrapper;
  let nextCb: jest.Mock;
  let prevCb: jest.Mock;
  beforeEach(() => {
    nextCb = jest.fn((value: CompareForm) => {});
    prevCb = jest.fn();
    wrapper = shallow(<StepAddress step={1} activeStep={1} onNext={nextCb} onPrev={prevCb} value={undefined} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("subbmit form", () => {
    wrapper.find('input[id="streetNumber"]').simulate("change", {
      target: {
        value: "3",
      },
    });
    wrapper.find('input[id="streetName"]').simulate("change", {
      target: {
        value: "nelson",
      },
    });
    wrapper.find('select[id="streetType"]').simulate("change", {
      target: {
        value: "Ave",
      },
    });
    wrapper.find('input[id="suburb"]').simulate("change", {
      target: {
        value: "indooroopilly",
      },
    });
    wrapper.find('input[id="postcode"]').simulate("change", {
      target: {
        value: "4068",
      },
    });
    wrapper.find('form[id="address-form"]').simulate("submit", { preventDefault() {} });
    expect(nextCb).toBeCalledWith({
      address: {
        streetNumber: "3",
        streetName: "nelson",
        streetType: "Ave",
        suburb: "indooroopilly",
        postcode: "4068",
      },
    } as CompareForm);
  });
});
