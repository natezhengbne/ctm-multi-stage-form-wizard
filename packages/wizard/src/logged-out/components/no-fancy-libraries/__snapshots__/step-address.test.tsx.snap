// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page of step-address renders renders correctly 1`] = `
<Fragment>
  <form
    id="address-form"
    onSubmit={[Function]}
  >
    <div
      className="StepperForm-root container-sm"
    >
      <h1>
        Address
      </h1>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="streetNumber"
        >
          Street Number *
        </label>
        <input
          className="form-control"
          id="streetNumber"
          max="10000"
          maxLength={16}
          min="1"
          name="streetNumber"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="streetName"
        >
          Street Name *
        </label>
        <input
          className="form-control"
          id="streetName"
          maxLength={16}
          name="streetName"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
        >
          Street Type *
        </label>
        <select
          className="form-select form-control"
          id="streetType"
          name="streetType"
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            disabled={true}
            key="Choose Street Type"
            value=""
          >
            Choose Street Type
          </option>
          <option
            key="CI"
            value="CI"
          >
            CI
          </option>
          <option
            key="Ct"
            value="Ct"
          >
            Ct
          </option>
          <option
            key="St"
            value="St"
          >
            St
          </option>
          <option
            key="PI"
            value="PI"
          >
            PI
          </option>
          <option
            key="Ave"
            value="Ave"
          >
            Ave
          </option>
        </select>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="suburb"
        >
          Suburb *
        </label>
        <input
          className="form-control"
          id="suburb"
          maxLength={30}
          name="suburb"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="postcode"
        >
          Postcode *
        </label>
        <input
          className="form-control"
          id="postcode"
          maxLength={20}
          name="postcode"
          onChange={[Function]}
          pattern="^(0[8][0-9]{2})|([1-7][0-9]{3})$"
          required={true}
          title="Postcode: 0800-7999"
          value=""
        />
      </div>
      <div
        className="StepperActions-root"
      >
        <button
          className="btn btn-secondary m-3"
          onClick={[Function]}
        >
          Back
        </button>
        <button
          className="btn btn-primary"
          type="submit"
        >
          Next
        </button>
      </div>
    </div>
  </form>
</Fragment>
`;
